io/subsystem/src/main/java/org/wildfly/extension/io/BufferPoolResourceDefinition.java

```diff
@@ -128,29 +119,46 @@ public void registerCapabilities(ManagementResourceRegistration resourceRegistra
         resourceRegistration.registerCapability(IO_POOL_RUNTIME_CAPABILITY);
     }
 
-    private static class BufferPoolAdd extends AbstractAddStepHandler {
+    @Override
+    public void registerAttributes(ManagementResourceRegistration resourceRegistration) {
+        resourceRegistration.registerReadOnlyAttribute(BUFFER_SIZE, new BufferReadAttributeHandler(BUFFER_SIZE));
```

Comment:
are they read-only? They are defined in the model and the user can define values when the buffer-pool resource is added.

Reply:
Yes, the attributes should be read-only, @ctomc mentioned that on the [upstream issue](https://issues.jboss.org/browse/WFCORE-2620?focusedCommentId=13406312&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-13406312) (if I understood correctly, of course).

Reply:
That's not what he said :)

I'm paraphrasing him as he said that there is no need to define a custom write operation handler as the buffer pool can not be dynamically resized.
However the attribute must still use a default write operation that will modify the model and put the server in reload-required (so that the new value will be only taken into account when the server is reloaded).

Reply:
Of course! I've jumped from "can't be change dynamically" to "readonly" but it's not the same. My bad. I'll change that too...